### PR TITLE
Added dimension model and used it to localize content

### PR DESCRIPTION
**Pull request by wachterjohannes**

### CODE REVIEW

File: DependencyInjection/SuluSyliusConsumerExtension.php [hunk 1]
```diff
@@ -103,15 +103,15 @@ public function prepend(ContainerBuilder $container)
...
                                 'path' => '%kernel.project_dir%/config/templates/products',
-                                'type' => 'product_content',
+                                'type' => 'products',
                             ],
                         ],
                         'resources' => [
                             'product_contents' => [
                                 'datagrid' => Product::class,
-                                'types' => ['product_content'],
+                                'types' => ['products'],
```
Comment (wachterjohannes) [inline DependencyInjection/SuluSyliusConsumerExtension.php]: should be named by the dependending resource-key

File: EventSubscriber/ContentSerializerSubscriber.php [hunk 2]
```diff
@@ -32,17 +33,39 @@ public static function getSubscribedEvents()
...
-        foreach ($object->getData() as $key => $value) {
-            $visitor->setData($key, $value);
+        foreach ($metadata->getProperties() as $property) {
+            if (array_key_exists($property->getName(), $data)) {
+                $visitor->setData($property->getName(), $data[$property->getName()]);
+
+                continue;
+            }
+
+            $visitor->setData($property->getName(), null);
```
Comment (wachterjohannes) [inline EventSubscriber/ContentSerializerSubscriber.php]: all properties have to be present

File: Model/Dimension/DimensionInterface.php [hunk 3]
```diff
@@ -15,6 +15,11 @@
 
 interface DimensionInterface
 {
+    const ATTRIBUTE_STAGE = 'stage';
```
Comment (chirimoya) [inline Model/Dimension/DimensionInterface.php]: ATTRIBUTE_KEY_STAGE

File: Model/Dimension/DimensionInterface.php [hunk 4]
```diff
@@ -15,6 +15,11 @@
...
+    const ATTRIBUTE_STAGE_DRAFT = 'draft';
```
Comment (chirimoya) [inline Model/Dimension/DimensionInterface.php]: ATTRIBUTE_VALUE_DRAFT

File: Model/Dimension/DimensionInterface.php [hunk 5]
```diff
@@ -15,6 +15,11 @@
...
+
+    const ATTRIBUTE_LOCALE = 'locale';
```
Comment (chirimoya) [inline Model/Dimension/DimensionInterface.php]: ATTRIBUTE_KEY_LOCALE

**Pull request merged**